js/zb.js

```diff
@@ -969,20 +969,49 @@ module.exports = class zb extends Exchange {
     }
 
     async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
-        if (type !== 'limit') {
-            throw new InvalidOrder (this.id + ' allows limit orders only');
-        }
         await this.loadMarkets ();
-        const request = {
-            'price': this.priceToPrecision (symbol, price),
-            'amount': this.amountToPrecision (symbol, amount),
-            'tradeType': (side === 'buy') ? '1' : '0',
-            'currency': this.marketId (symbol),
-        };
-        const response = await this.spotV1PrivateGetOrder (this.extend (request, params));
+        const market = this.market (symbol);
+        if (market['spot']) {
+            if (type !== 'limit') {
+                throw new InvalidOrder (this.id + ' allows limit orders only');
+            }
+        }
+        const method = this.getSupportedMapping (market['type'], {
+            'spot': 'spotV1PrivateGetOrder',
+            'swap': 'contractV2PrivatePostTradeOrder',
+        });
+        const request = {};
```

Comment:
const request = {
            'amount': this.amountToPrecision (symbol, amount);
        };
        if (price) {
            request['price'] = this.priceToPrecision (symbol, price);
        }
        if (market['spot']{
            request['tradeType'] = (side === 'buy') ? '1' : '0';
            request['currency'] = market['id'];
        } else if (market['swap']) {
               ...
        }